packages/phone-number-privacy/src/match-making/get-contact-matches.ts

```diff
@@ -1,26 +1,30 @@
 import { Request, Response } from 'firebase-functions'
 import { ErrorMessages, respondWithError } from '../common/error-utils'
 import { authenticateUser } from '../common/identity'
+import { getDidMatchmaking, setDidMatchmaking } from '../database/wrappers/account'
 import { getNumberPairContacts, setNumberPairContacts } from '../database/wrappers/number-pairs'
 
 // TODO (amyslawson) consider pagination or streaming of contacts?
 export async function handleGetContactMatches(request: Request, response: Response) {
-  // TODO (amyslawson) refactor this internal logic and input validation to its own file
-  // when adding error handling
   try {
     if (!isValidGetContactMatchesInput(request.body)) {
       respondWithError(response, 400, ErrorMessages.INVALID_INPUT)
       return
     }
     authenticateUser()
+    if (await getDidMatchmaking(request.body.account)) {
+      console.warn(ErrorMessages.DUPLICATE_REQUEST_TO_MATCHMAKE)
+      respondWithError(response, 403, ErrorMessages.DUPLICATE_REQUEST_TO_MATCHMAKE)
+      return
+    }
     const matchedContacts: ContactMatch[] = (
       await getNumberPairContacts(request.body.userPhoneNumber, request.body.contactPhoneNumbers)
     ).map((numberPair) => ({ phoneNumber: numberPair }))
     await setNumberPairContacts(request.body.userPhoneNumber, request.body.contactPhoneNumbers)
-    // TODO (amyslawson) return salts with contact
+    await setDidMatchmaking(request.body.account)
     response.json({ success: true, matchedContacts })
   } catch (e) {
-    console.error('Failed to getContactMatches', e)
+    console.error(ErrorMessages.UNKNOWN_ERROR + ' Failed to getContactMatches', e)
```

Comment:
With console, you can just provide multiple arguments. No need to `+` them together yourself

Reply:
k